src/net_processing.cpp

```diff
@@ -3490,13 +3490,15 @@ bool ProcessMessage(CNode& pfrom, const std::string& msg_type, CDataStream& vRec
     }
 
     if (msg_type == NetMsgType::FILTERCLEAR) {
+        if (!(pfrom.GetLocalServices() & NODE_BLOOM)) {
+            pfrom.fDisconnect = true;
+            return false;
```

Comment:
in commit 294b96772b7d184a0cd6f56bd76ef118f4c78cc6:

Any reason to return false here? The only effect is that it will log a line with `FAILED` with the NET debug category. However, it should be pretty clear from the log already that receiving a filterclear message and then disconnecting a peer means the message was ignored.

Reply:
I was confused here too, but it seems like this is a common thing to do. Many occurences of 
``
             pfrom.fDisconnect = true;
            return false; 
``

Reply:
The majority of disconnects is followed by a `return true;`. The ones that don't should be changed too. And longer term the return value should be removed completely.

Reply:
Thanks! Yup, my reasoning was that's what it looked like in other places 😅